### PR TITLE
Updating MySQL versions and adding features

### PR DESCRIPTION
**Pull request by whimxiqal**
- MySQL and MariaDB drivers have diverged a bit and require separate drivers for MySQL storage type
- Added a Condition for queries to allow specification for a general prismatic region (API)
- Moved "ApplierCommand" logic to a separate named function (API)

### CODE REVIEW

File: src/main/java/com/helion3/prism/storage/mysql/MySQLStorageAdapter.java [hunk 1]
```diff
@@ -53,7 +53,7 @@
     public MySQLStorageAdapter() {
         records = new MySQLRecords();
 
-        dns = String.format("jdbc:mysql://%s/%s",
+        dns = String.format("jdbc:mysql://%s/%s?useSSL=false",
```
Comment (viveleroi) [inline src/main/java/com/helion3/prism/storage/mysql/MySQLStorageAdapter.java]: What's the purpose of this? Feels like something we shouldn't be hard-coding to false unless. I'm fine if we default it false in the config though.

**Review by viveleroi — COMMENTED** [review]
Thanks! Looks good overall.

Do you know why a merge commit is showing in the list of commits though? I'd prefer to avoid merge commits unless they represents merges in this master. Maybe a rebase can clean that out or something.

File: src/main/java/com/helion3/prism/storage/mysql/MySQLStorageAdapter.java (hunk 1 above)
Comment (whimxiqal) [reply src/main/java/com/helion3/prism/storage/mysql/MySQLStorageAdapter.java]: Regarding the merge commit, I tried to get rid of it but it's not something I can easily do it seems. The merge is with the master repo here though, so it doesn't affect anything. I can try any ideas you have on how to get rid of it.

Comment (whimxiqal) [reply src/main/java/com/helion3/prism/storage/mysql/MySQLStorageAdapter.java]: I'll add the SSL setting to the config. I think I just had that in there to get SQL Workbench to work on my laptop but I might not need it at all

**Comment by whimxiqal** [comment]
Can you see why this build is failing? I mistakenly removed some of the gradle files but I restored them and something doesn't work anymore.

**Comment by viveleroi** [comment]
It's failing because of gradle issues: `./gradlew: Permission denied`

https://travis-ci.org/github/prism/Prism/jobs/666203178?utm_medium=notification&utm_source=github_status

**Comment by phit** [comment]
specifically you changed the file permissions of the script to be non executable, you can reset it with
`git checkout 1ed5796 -- gradlew`

or just chmod if you are on linux

**Comment by whimxiqal** [comment]
Alright, permissions fixed and SSL qualifier was removed. I don't think a config setting is really necessary for the SSL specification and my local database works fine with just the new drivers.

File: .gitignore [hunk 2]
```diff
@@ -65,6 +65,9 @@ Thumbs.db
 .project
 .settings
 eclipse
+gradle
```
Comment (phit) [inline .gitignore]: forgot to remove this? they are part of the repo so doesnt make much sense to gitignore them

Comment (whimxiqal) [reply .gitignore]: Yes, I don't know why the .gitignore wasn't committed along with my other files.

**Comment by viveleroi** [comment]
Looks fine now, ready for me to squash and merge?

**Comment by whimxiqal** [comment]
ready

**Pull request merged**